libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.java

```diff
@@ -151,6 +153,18 @@ protected void onNewIntent(Intent intent) {
 
         // Reloads login page for every new intent to ensure the correct login server is selected.
         webviewHelper.loadLoginPage();
+
+        // Launches IDP login flow directly for IDP initiated login flow.
+        if (intent != null) {
+            final Bundle extras = intent.getExtras();
+            if (extras != null) {
+                userHint = extras.getString(IDPInititatedLoginReceiver.USER_HINT_KEY);
+                boolean isIdpInitFlow = extras.getBoolean(IDPInititatedLoginReceiver.IDP_INIT_LOGIN_KEY);
+                if (isIdpInitFlow) {
```

Comment:
If this is an IDP initiated auth request, take `user_hint` and directly flow back to the IDP app.